src/ui/components/NetworkMonitor/index.tsx

```diff
@@ -87,31 +87,32 @@ export const NetworkMonitor = ({
 
         return (
           <div className="flex h-full min-h-0 flex-col" ref={container}>
-            <FilterBar types={types} toggleType={toggleType} table={table} />
-            <SplitBox
-              className="min-h-0 border-t border-splitter"
-              initialSize="350px"
-              minSize={selectedRequest ? "30%" : "100%"}
-              maxSize={selectedRequest ? "70%" : "100%"}
-              startPanel={
-                <RequestTable
-                  table={table}
-                  data={data}
-                  currentTime={currentTime}
-                  onRowSelect={row => {
-                    trackEvent("net_monitor.select_request_row");
-                    dispatch(selectAndFetchRequest(row.id));
-                  }}
-                  seek={seek}
-                  selectedRequest={selectedRequest}
-                />
-              }
-              endPanel={
-                selectedRequest ? <RequestDetails cx={cx} request={selectedRequest} /> : null
-              }
-              splitterSize={2}
-              vert={vert}
-            />
+            <Filter types={types} toggleType={toggleType} table={table}>
+              <SplitBox
```

Comment:
Opted for composition here so when `Filter` is opened/closed it doesn't re-render this component.

Reply:
Nit: this confused me a bit since the Filter component is more like a wrapper. Is there another way to avoid the re-render?

Reply:
Yeah, this is a little weird with `children` prop and makes it harder to reason about. I like the drawer component idea you mention below, we had this at my previous job and called it `FilterLayout` which might work here. We can use slots that handle opening/closing the sidebar:

```
<FilterLayout
  sidebarWidth={224}
  sidebar={<div>Sidebar</div>}
  toolbar={<div>Toolbar</Box>}
  table={<div>Table</div>}
/>
```